models/repo/repo_list.go

```diff
@@ -745,3 +745,34 @@ func GetUserRepositories(opts *SearchRepoOptions) (RepositoryList, int64, error)
 	repos := make(RepositoryList, 0, opts.PageSize)
 	return repos, count, db.SetSessionPagination(sess, opts).Find(&repos)
 }
+
+func GetPrimaryRepoLanguageList(ctx context.Context, ownerID int64, private bool) (LanguageStatList, error) {
+	languageList := make(LanguageStatList, 0)
+
+	q := db.GetEngine(ctx).
+		Table("language_stat").
+		Cols("language").
+		Where(builder.Eq{"is_primary": true})
+
+	if ownerID > 0 {
+		ids, err := SearchRepositoryIDsByCondition(ctx, builder.NewCond().And(
+			builder.Eq{"owner_id": ownerID},
+			builder.Eq{"is_private": private},
```

Comment:
This will not work as you expect. You can give other users access to a private repo. if you do this, they will see the private repo. So you need to check for each repo if the current user has read access. Take a look at  `repo_model.SearchRepositoryByName` and `repo_model.CountRepository` which both o the check.

Reply:
Eek, I should have tested out with multiple accounts